source/transfer_list.rst

```diff
@@ -181,6 +181,99 @@ The TE header is defined in :numref:`tab_te_header`.
      - The exact size of the data content in bytes, not including inter-TE padding. May be 0.
 
 
+Entry-type allocation
+---------------------
+
+Tag IDs must be allocated in this specification before use. A new tag ID can be
+allocated by submitting a pull request to this repository that adds a
+description of the respective TE data layout to this specification. Tag IDs do
+not have to be allocated in order. Submitters are encouraged to try to group
+tag IDs together in logical clusters at 16 or 256-aligned boundaries (e.g. all
+tags related to a particular chipset or to a particular firmware project could
+use adjacent tag numbers), but there are no predefined ranges and no
+reservations of tag ranges for specific use.
+
+Tags are expected to have a simple layout (representable by a C structure) and
+each tag should only represent data for a single logical concept. Data for
+multiple distinct concepts should be split across different tags, even if
+they're always expected to appear together on the first platform adding the tag
+(to encourage reusability in different situations). Alternatively, complex data
+may be represented in a different kind of well-established handoff data
+structure (e.g. FDT [DT]_, HOB [PI]_) that is inserted into the TL as a single
+TE. The same tag ID may occur multiple times in the TL to represent multiple
+instances of the same kind of object. Tag layouts (including the meaning of all
+fields) are considered stable after being added to this specification and may
+never be changed in a backwards-incompatible way. If a backwards-incompatible
+change is desired, a new tag ID should be allocated for the new version of the
+layout instead.
+
+Tag layouts may be changed in a backwards-compatible manner by allowing new
+valid values in existing fields (including reserved fields), as long as the
+original layout definition clearly defined how unknown values in those fields
+should be handled, and the rest of the TE would still be considered valid and
+correct for older readers that consider the new values unknown. TE layouts may
+also be expanded by adding new fields at the end, with the same restrictions.
+TEs should not contain explicit version numbers and instead just use the
+`data_size` value to infer how many fields exist. TE layouts which have been
+changed like this must clearly document which fields or valid values were added
+at a later time, and in what order.
```

Comment:
...to here would fit better under the "Tag Contents" umbrella, if we're introducing that as a separate section. (Also it would probably look better to have Tag Contents before Tag Allocation in that case.)